test/helpers.py

```diff
@@ -48,3 +51,31 @@
 @eqx.filter_jit
 def make_mlp(key):
     return eqx.nn.MLP(2, 2, 2, 2, key=key)
+
+
+@contextlib.contextmanager
+def assert_no_garbage(
+    allowed_garbage_predicate: Callable[[Any], bool] = lambda _ : False
+):
+    try:
+        gc.disable()
+        gc.collect()
+        gc.collect()
```

Comment:
Note - I don't understand why the second collect() is necessary here, but it seems like it is...!

Reply:
How mysterious! Can you add a comment so we don't remove it later?

Reply:
Done